### ISSUE TITLE
[Bug] grafana 4.5.x and graphite-api ( not default graphite )

### ISSUE THREAD

**Issue by derkgort** [grafana/grafana#9258]
I'm getting this error in the webinterface after the update from 4.4.3 to 4.5.0. Tried 4.5.1 too. Same issue
I think the new version and the check for 1.0 vs 0.9 of graphite made this pop-up.

graphite plugin failed
TypeError: Cannot read property 'graphiteVersion' of undefined

Graphite API is a api only alternative to graphite web. We use it because it's leaner and easier to configure for multiple carbon directories. I think it does not have the params/json content that grafana is looking for.

Please include this information:
- What Grafana version are you using?
4.5.0, 4.5.1
- What datasource are you using?
graphite with graphite-api
- What OS are you running grafana on?
centos 7 latests
- What did you do?
usage graphite-api, not graphite-web and update from 4.4 to 4.5

- What was the expected result?
Still working

- What happened instead?
Got the error

**Comment by torkelo** [comment]
When / where do you get the error?

**Comment by bergquist** [comment]
https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/graphite/datasource.ts#L14 are we sure the datasource have jsonData?

**Comment by torkelo** [comment]
It should not be null, empty object maybe

**Comment by bergquist** [comment]
yea. Me and @daniellee is trying to reproduce this.

I have the same error last week but now I cannot reproduce it.

**Comment by derkgort** [comment]
I get the error when i try to open grafana in the browser. Linux Chrome.
I had some tabs open and those still work.

I've downgraded back to 4.4 and it worked again.

**Comment by derkgort** [comment]
And i have multiple orgs that connect to the different api's running on different ports.

**Comment by azhiltsov** [comment]
Seems updater during update does not update json_data for graphite datasources. Can be manually fixed by updating each graphite datasource from the web-interface or by running SQL:
```
update data_source set json_data='{"graphiteVersion":"0.9"}' where type='graphite' and json_data='{}'
```

**Comment by daniellee** [comment]
Can recreate this now. If the json_data field in the data_source table is null, then you get this error. When we first tried to recreate the error and failed, the json_data field was empty string rather than null.

This [line](https://github.com/grafana/grafana/blob/master/pkg/api/frontendsettings.go#L63-L65) in the frontendsettings api code, skips adding the json_data field if the value is null. A simple fix would be to have an else clause here and create an empty object.

**Comment by derkgort** [comment]
thanks for the attention to this issue!

**Comment by rotanid** [comment]
thanks @azhiltsov for the hint, that fixes the issue!

**Comment by torkelo** [comment]
a temp workaround until patch release is out should be going into data source edit view and hit update

**Comment by azhiltsov** [comment]
I attempted to migrate from 4.4.3 to 4.5.2 and seems the issue with wrong values in the json_data column is still there.
I am still getting "TypeError: Cannot read property 'graphiteVersion' of null" on dashboard load 
And json_data is either null or {}
```
mysql> select json_data from data_source where type = 'graphite';
+-----------+
| json_data |
+-----------+
| null      |
| null      |
| null      |
| {}        |
| {}        |
| {}        |
| {}        |
| {}        |
| {}        |
+-----------+
9 rows in set (0.01 sec)
```
In migration_log there is only one new record: "alter table `user` ADD COLUMN `last_seen_at` DATETIME NULL"

**Comment by torkelo** [comment]
we did not solve it via sql migration, but when we read the data and send to the frontend. 

When do you get the error ? if you do "view source"  (read html, there should be a json blob at the end with your data sources, what is the property of jsonData ?

**Comment by azhiltsov** [comment]
As mentioned earlier - on dashboard load. Basically all dashboards with graphite datasoure became broken after update.

**Comment by torkelo** [comment]
strange, and your sure your running 4.5.2 ? (btw, the same fix for 4.5.1 should still work, go into data source edit view and save datasource, do a full page reload)

**Comment by lathiat** [comment]
Also having this problem on 4.5.2 -- the data source edit/save fixed it but it did not fix itself.

**Comment by davidlmaldonado** [comment]
This worked for me. Upgraded from grafana-4.4.3 to grafana-4.5.2.

**Comment by alaphoid** [comment]
I upgraded from 4.4.3 to 4.5.2 and had to edit each data source and click 'Save & Test'.  Before doing this I was getting the error: "TypeError: Cannot read property 'graphiteVersion' of null" when loading a dashboard.

**Comment by hanynowsky** [comment]
I got the same issue saying a.JsonData null when upgraded from 4.2 to 4.5. The problem is that I had beforehand setup the home page to be a specific dashboard. SO no way to access whatever menu for you get a blank page.
Anyway, downgraded to 4.2, then set the home page to be the default, then edited each datasource. Upgraded again to 4.5 and voilà, it's working as torkelo suggested. Hopefully this should be fixed in 4.6. Thanks for your feedback.
